### DISCUSSION TITLE
Saucedacity 1.1

### DISCUSSION THREAD

**Discussion by generic-pers0n** [saucedacity/saucedacity#17] [Announcements]
Today marks the release of Saucedacity 1.1. However, on this same day, there was an attack on the Twin Towers in the United States. Many people died in that attack, leading to a string of events from the attacks themselves to many others, some of which have recent effects as of today.

There are various coincidences with the 20th anniversary of 9/11 and the Saucedacity 1.1 release today. One coincidence is the release date itself. The second coincidence is the 1.1 version number, which came naturally (I didn't predict Saucedacity 1.1 being released on 9/11). However, I wouldn't call them complete coincidences. While I did not expect Saucedacity 1.1 to be released officially on 9/11, I would like to bring something special to the table. I would like to dedicate this release of Saucedacity to those who died on 9/11, those who died from illnesses caused by 9/11, the survivors of 9/11, the first responders of 9/11, basically everyone that was involved. There's just too much to list here, but for those involved, I dedicate this release to you for everything. You've been through so much, regardless of who you are or your actual role, in so many different ways.

With that said, lets get to the changelog of this release! Nothing too special, but here's what we have anyways:

# New
* A new compile-time option has been introduced, `-Dsaucedacity_enabled_legacy`. This disables some legacy options for now, namely LADSPA plugin support (in addition to dealing with extremely old Audacity project formats). Right now, this is enabled by default, but will be disabled in other releases.
* Saucedacity now recognizes if its preferences were written by Saucedacity or Audacity (`IsSaucedacity` in the preferences). This means nothing but is more or less for future proofing. You will be notified if these preferences were not made by Saucedacity, although clicking 'OK' will only mean setting the `IsSaucedacity` preference to `1`, being true.

# Changes
* The welcome dialog now only shows up on first run (unless Saucedacity's preferences are reset or deleted).
* More branding changes, including a new splash screen logo and some interim branding.
* We're now based on Audacity 3.0.4, incorporating Audacity's latest changes (minus the error reporting and update checking).
* The Linux binaries were built on Ubuntu 18.04 rather than Pop!_OS 21.04. This comes from #16, largely in order to increase binary compatibility among different Linux distros.

Anyways, that's it for this release. Hope you like this release!

_To all the 9/11 victims, survivors, families, and anyone else involved, this release is for you. I hope to bring something to your hearts, but this won't compensate for the damage caused by 9/11, be it physical, emotional, both, or in any other way. Nevertheless, I shall pay all my respects to you. For first responders, we thank you for your service. For victims and their families, I pay my respects to you._

_- The Saucedacity Team, with love and support all the way._

<hr /><em>This discussion was created from the release <a href='https://github.com/saucedacity/saucedacity/releases/tag/v1.1'>Saucedacity 1.1</a>.</em>

**Comment by gotsum** [comment]
Hey, Sauce team. I'm a real casual user--not too techie, but I can act like one to stay as FOSS as possible, learning as I go.
Just now realized that I had been using Audacity 2.2.1 before shifting to Sauce. I didn't know that saving a project without "printing to disc" (to use Pro Tools jargon) was disabled in future Audacity releases. I chatted about it on the Audacity forums here: https://forum.audacityteam.org/viewtopic.php?f=48&t=121925
It would be mega cool if Saucedacity could bring this feature back!!! By "printing to disc", I mean that the files save the entire audio data. No longer does Audacity have the option to only save the "data map" (or whatever to call it), but I relied on it and unwittingly had never been updating Audacity.

So my post is twofold:
(1) If it's possible and valuable to yous guys, restoring this feature would be a dream come true!
(2) Does using an old version of Audacity, and never updating it, suffice to block the privacy-invasive "features" that precipitated making forks? Forgive my ignorance.
-
Besides that, I can report the following as a very frequent though simple user of Saucedacity:

When I have a lot of tracks, the view tends to cut off the bottom of the last track, and also can suddenly shift the view upwards when I click on the file in that track. The workaround is to just line the cursor up carefully, dbl click it and then I can cut/paste it somewhere else (where the view remains untainted).

Also, I frequently wish I know of a way to open Sauce without having to drag the "run" file into terminal (nor use terminal at all). How do you open Saucedacity in the easiest way?

Similar to that last point, it would be fantastic if one could see Saucedacity pop up in "open file with..." right clicks on audio files, the way Audacity does.

Hope the latter info helps your efforts, and that you can let me know if simply using version 2.2.1 of Audacity is unadvisable for being telemetry free (or any other reason).

Thanks!

**Reply by generic-pers0n** [reply]
Hello @gotsum!

To answer some questions of yours, I'll be looking into adding that feature. Unfortunately, it will not be added in the (somewhat) upcomming 1.2 release (as I already have a release agenda for 1.2), but I'm kind of interested in exploring quite a range of things regarding this.

Second, answering your 2nd question, it does suffice as a way to block any of those provacy-invasive "features" (as you described). Of course, there's the obvious (missing new features, new security updates, bug fixes, etc), but if you're fine with that, then it does suffice as a way to block the privacy invasive features. Finally, this question wasn't an ignorant one; your question is quite a good one. If you find it desirable to kewo on using an old version of Audacity, then that's fine! If you'd rather use a newer fork, that's also fine too! Whatever your desire is, man! It's all you on that.

Regarding your report about the tracks, that's another thing that should be looked into. I can't guarantee any sort of changes will make it into 1.2 (as described earlier), but I'll also take a look into this. Thank you very much for reporting this.

Finally, about running Saucedacity from the archive without dragging the run.sh script into the terminal each time, you can probably make a synlink in /usr/bin to the run.sh script. You would still have to run it from the terminal, however, unless you create a .desktop file (maybe in /usr/share/applications system wide or ~/.local/share/applications for the current user only, if I get that correctly). You might be able to use the one from the source code (located in the 'src' folder). As for an "Open with Saucedacity" feature, I believe that's more towards the system's preferences, but still, I'll also look into this.

Overall, thank you for your suggestions and reports! I appreciate the time you took to make your post! I hope to make progress on any if these features, and I might have my own inspirations from these feature suggestions too! :)

**Reply by gotsum** [reply]
Thank you for your thorough answers, my friend.
I did a comparison by creating and saving the exact same thing in my old Audacity program (.aup) and Saucedacity (.aup3). Though I didn't realize the data folder for the .aup file was strangely larger than I thought it needed to be (measuring about 18 MB when the only audio file I imported into the project was about 4 MB--but maybe that's because it upsampled the source MP3 to a Wav), it is still about half as large as the .aup3 file (measuring in at 34MB). For me, file size is important because I usually make mixes all day that often use the same source song files. Maybe I'll get a bigger hard drive soon, though, and just commit to the new.

The world needs more FOSS DAW options in general. With all your knowledge, I bet you could make one that was badass!

**Reply by generic-pers0n** [reply]
Honestly, that's...something I didn't notice. I never paid any attention to file sizes, but that's quite an observation.

I see how working on many projects, this can add up. It would be a little interesting to study this further. But maybe if Saucedacity supported saving to the AUP file format again, that might be one thing.

Either way, this is all stuff I have to look into. Thanks for bringing this up too! :grinning:
